sqs/provider.go

```diff
@@ -46,6 +48,10 @@ type Provider struct {
 	Logger              lager.Logger
 }
 
+type BindParameters struct {
```

Comment:
non-blocking / consideration: Seems like this BindParameters type might be unneccasary and we could use the UserParams as the bind params (as we do with the QueueParams being the provision params). You can take advantage of `json:"-"` to ignore fields.